tests/integration/test_schema_registry_auth.py

```diff
@@ -0,0 +1,215 @@
+"""
+karapace - schema registry authentication and authorization tests
+
+Copyright (c) 2022 Aiven Ltd
+See LICENSE for details
+"""
+from karapace.client import Client
+from karapace.kafka_rest_apis import KafkaRestAdminClient
+from karapace.schema_models import SchemaType, ValidatedTypedSchema
+from tests.utils import (
+    new_random_name,
+    new_topic,
+    schema_avro_json,
+    schema_jsonschema_json,
+    test_objects_avro,
+    wait_for_topics,
+)
+from typing import List
+from urllib.parse import quote
+
+import aiohttp
+import asyncio
+import requests
+
+NEW_TOPIC_TIMEOUT = 10
+
+admin = aiohttp.BasicAuth("admin", "admin")
+aladdin = aiohttp.BasicAuth("aladdin", "opensesame")
+reader = aiohttp.BasicAuth("reader", "secret")
+
+
+async def test_sr_auth(registry_async_client_auth: Client) -> None:
+    subject = new_random_name("cave-")
+
+    res = await registry_async_client_auth.post(f"subjects/{quote(subject)}/versions", json={"schema": schema_avro_json})
+    assert res.status_code == 401
```

Comment:
Most of the existing tests use numbers for HTTP error codes instead of `HTTPStatus`, so maybe this needs general guideline which way we want it to be?

Reply:
Ok, if so let's ignore this for now. (I vote for named constants over literals.:)